apis/pipeline/pipeline.yaml

```diff
@@ -5390,6 +5566,77 @@ components:
                 mapping:
                     'pke-on-azure': '#/components/schemas/CreatePKEOnAzureClusterRequest'
 
+        NodePool:
+            oneOf:
+                - $ref: '#/components/schemas/EksNodePool'
+
+        GenericNodePool:
+            description: Generic node pool object for all cluster distributions.
+            type: object
+            required:
+                - name
+                - size
+            properties:
+                name:
+                    description: Node pool name.
+                    type: string
+                size:
+                    description: Node pool size.
+                    type: integer
+                labels:
+                    description: Node pool labels.
+                    type: object
+                    additionalProperties:
+                        type: string
+                    example:
+                        example.io/label1: value1
+
+        NodePoolAutoScaling:
+            description: Node pool auto scaling settings.
+            type: object
+            required:
+                - minSize
+                - maxSize
+            properties:
+                enabled:
+                    description: Enable node pool autoscaling.
+                    type: boolean
+                minSize:
+                    description: Minimum node pool size.
+                    type: integer
+                    minimum: 0
+                    example: 1
+                maxSize:
+                    description: Maximum node pool size.
+                    type: integer
+                    example: 5
+
+        EksNodePool:
+            description: Node pool object for an EKS cluster.
+            type: object
+            allOf:
+                - $ref: '#/components/schemas/GenericNodePool'
+                -
+                    type: object
+                    required:
+                        - instanceType
+                    properties:
+                        autoscaling:
+                            $ref: '#/components/schemas/NodePoolAutoScaling'
+                        instanceType:
+                            description: Machine instance type.
+                            type: string
+                            example: "m4.xlarge"
+                        image:
+                            description: Instance AMI.
+                            type: string
+                            example: "ami-06d1667f"
+                        spotPrice:
+                            description: Spot instance price.
+                            type: string
+                            example: "0.2"
```

Comment:
```suggestion
                            example: "0.2"
                            description: "The upper limit price for the requested spot instance. If this field is left empty or 0 passed in on-demand instances used instead of spot instances "
```